src/phyltr/commands/base.py

```diff
@@ -1,28 +1,75 @@
 import fileinput
 import shlex
 import sys
+import argparse
+import re
 
 from phyltr.plumbing.sources import NewickParser
 from phyltr.plumbing.sinks import NewickFormatter
-from phyltr.utils.phyltroptparse import OptionParser
+
 
 class PhyltrCommand(object):
 
-    parser = OptionParser("Halp!")
     source = NewickParser
     sink = NewickFormatter
+    __options__ = []  # Derived classes should add parser arguments here.
+    __opt_dests = []  # When initializing an ArgumentParser, we store the argument dests here.
+
+    def __init__(self, _opts=None, **kw):
+        """
+        Command options can be initialized in two ways:
+
+        1. By passing an `argparse.Namespace` object as `_opts`.
+        2. By passing values for individual options a keyword arguments.
+        """
+        if _opts:
+            self.opts = _opts
+        else:
+            # We initialize the options with the default values of the ArgumentParser:
+            p = self.parser()
+            defaults = {d: p.get_default(d) for d in self.__opt_dests}
+            defaults.update(kw)
+            self.opts = argparse.Namespace(**defaults)
```

Comment:
This block makes sure, non-command line usage also leads to a fully populated `opts` namespace.